OpenMEEG/src/gain.cpp

```diff
@@ -113,17 +107,9 @@ int main(int argc, char **argv)
             cerr << "Not enough arguments \nPlease try \"" << argv[0] << " -h\" or \"" << argv[0] << " --help \" \n" << endl;
             return 0;
         }
-        LinOpInfo matinfo = OpenMEEG::maths::info(argv[3]);
-        SymMatrix HeadMatInv;
-        HeadMatInv.load(argv[2]);
-        Matrix SourceMat;
-        SourceMat.load(argv[3]);
-        Matrix Head2MEGMat;
-        Head2MEGMat.load(argv[4]);
-        Matrix Source2MEGMat;
-        Source2MEGMat.load(argv[5]);
+        const Matrix &tmp = Matrix(argv[5]) + (Matrix(argv[4]) * SymMatrix(argv[2])) * Matrix(argv[3]);
 
-        GainMEG MEGGainMat(HeadMatInv, SourceMat, Head2MEGMat, Source2MEGMat);
+        GainMEG MEGGainMat(tmp);
```

Comment:
because each matrix is being released after the computation of the product or addition